src/khoj/routers/helpers.py

```diff
@@ -961,17 +964,67 @@ async def text_to_image(
                         "aspect_ratio": "1:1",
                     },
                 )
-                decoded_image = response.content
+                # Convert png to webp for faster loading
+                webp_image_bytes = convert_image_to_webp(response.content)
             except requests.RequestException as e:
                 logger.error(f"Image Generation failed with {e}", exc_info=True)
                 message = f"Image generation failed with Stability AI error: {e}"
                 status_code = e.status_code  # type: ignore
                 yield image_url or image, status_code, message, intent_type.value
                 return
 
-    with timer("Convert image to webp", logger):
-        # Convert png to webp for faster loading
-        webp_image_bytes = convert_image_to_webp(decoded_image)
+    elif text_to_image_config.model_type == TextToImageModelConfig.ModelType.REPLICATE:
+        with timer("Generate image using Replicate", logger):
+            try:
+                # Create image generation task on Replicate
+                create_prediction_url = f"https://api.replicate.com/v1/models/{text2image_model}/predictions"
+                headers = {
+                    "Authorization": f"Bearer {text_to_image_config.api_key}",
+                    "Content-Type": "application/json",
+                }
+                json = {
+                    "input": {
+                        "prompt": improved_image_prompt,
+                        "num_outputs": 1,
+                        "aspect_ratio": "1:1",
+                        "output_format": "webp",
+                        "output_quality": 100,
+                    }
+                }
+                create_prediction = requests.post(create_prediction_url, headers=headers, json=json).json()
+
+                # Get status of image generation task
+                get_prediction_url = create_prediction["urls"]["get"]
+                get_prediction = requests.get(get_prediction_url, headers=headers).json()
+                status = get_prediction["status"]
+                retry_count = 1
+
+                # Poll the image generation task for completion status
```

Comment:
Man, that's rough that we have to poll it. Wouldn't it block the whole thread? How long does an image take, approx? If it' just like 10-15s, it's manageable, but if it's like 45s we might have to think about a better way to do this. I suppose we could keep sending updates to the client as well.

Reply:
Yeah, the polling required for the replicate API is annoying. Time depends on the model but haven't seen image generation to take more than 15s. E.g flux schnell takes ~3s, flux dev takes ~6s and flux pro takes ~15s in my testing